Utils.py

```diff
@@ -385,6 +386,43 @@ def persistent_load() -> typing.Dict[str, dict]:
     return storage
 
 
+def load_data_package_for_checksum(game: str, checksum: typing.Optional[str]):
+    def is_filename_safe(s):
+        return "/" not in s and "\\" not in s and ":" not in s
+
+    if checksum and is_filename_safe(game) and is_filename_safe(checksum):
+        path = user_path(os.path.join("datapackage", game, f"{checksum}.json"))
+        if os.path.exists(path):
+            try:
+                with open(path, "r", encoding="utf-8-sig") as f:
+                    return json.load(f)
+            except Exception as e:
+                logging.debug(f"Could not load data package: {e}")
+
+    # fall back to old cache
+    cache = persistent_load().get("datapackage", {}).get("games", {}).get(game, {})
+    if cache.get("checksum") == checksum:
+        return cache
+
+    # cache does not match
+    return {}
+
+
+def store_data_package_for_checksum(game: str, data: typing.Dict[str, Any]):
+    def is_filename_safe(s):
```

Comment:
so if a game name has a : for example in its title, then it won't be cached?

Reply:
oh, the idea was to have the game verbatim. we can probably sanitize the path until before hash and then only abort on bad symbols in the hash. i didn't find the function to construct a clean path you mentioned.